### PR TITLE
Expose sign function taking committed message vector

### PR DESCRIPTION
**Pull request by ale-linux**
The new function, `SignWithKeyB` signs a pre-generated `b` term.

### CODE REVIEW

File: bbs/bbs_test.go [hunk 1]
```diff
@@ -262,3 +263,83 @@ func TestBBSG2Pub_DeriveProof(t *testing.T) {
 			"larger than 4 messages")
 	})
 }
+
+// TestBlindSign uses `SignWithKeyB` to show how blind singing could be implemented
```
Comment (yacovm) [inline bbs/bbs_test.go:267]: Singing?

File: bbs/bbs_test.go [hunk 2]
```diff
@@ -262,3 +263,83 @@ func TestBBSG2Pub_DeriveProof(t *testing.T) {
...
+		}
+
+		cb.Add(pubKeyWithGenerators.H[i], bbs.FrFromOKM(msg))
+	}
+	cb.Add(b_req, curve.NewZrFromInt(1))
+	cb.Add(curve.GenG1, curve.NewZrFromInt(1))
+
+	// signer signs
+	scheme := bbs.New()
+	sig, err := scheme.SignWithKeyB(cb.Build(), len(messagesBytes), privKey)
```
Comment (yacovm) [inline bbs/bbs_test.go:332]: cb.build() should probably be exported to before //signer signs

File: bbs/bbs_test.go (hunk 1 above)
Comment (ale-linux) [reply bbs/bbs_test.go:267]: :man_facepalming:  fixed

File: bbs/bbs_test.go (hunk 2 above)
Comment (ale-linux) [reply bbs/bbs_test.go:332]: good point - done

**Comment by ale-linux** [comment]
Thanks @yacovm I've addressed all your comments

**Review by yacovm — APPROVED** [review]

**Pull request merged**